Lib/fontTools/feaLib/variableScalar.py

```diff
@@ -83,29 +83,37 @@ def default(self):
             # I *guess* we could interpolate one, but I don't know how.
         return self.values[key]
 
-    def value_at_location(self, location):
+    def value_at_location(self, location, avar=None):
         loc = location
         if loc in self.values.keys():
             return self.values[loc]
         values = list(self.values.values())
-        return self.model.interpolateFromMasters(loc, values)
+        return self.model(avar).interpolateFromMasters(loc, values)
 
-    @property
-    def model(self):
+    def model(self, avar=None):
         key = tuple(self.values.keys())
         if key in self.model_pool:
             return self.model_pool[key]
```

Comment:
if the VariableScalar.values that are used as the key in the model_pool chace are user-space, pre-normalization.. but then you pass in avar as a parameter, which in turn might change the normalized location stored inside the VariationModel...
the cache's key should either include avar somehow, or it should have normalized values to be avar-independent, otherwise there's a risk that you get a wrong cached model, one that doesn't take the passed-in avar into account.

Reply:
Ugh. Maybe (store into/load from) cache first, then apply avar? What's a good benchmark for ensuring this doesn't undo the optimization?

Reply:
how about we do not store the model cache inside the VariableScalar class scope _at all_? Can you try moving it to the feaLib builder's state?